tasks/create-github-release.yaml

```diff
@@ -0,0 +1,114 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: create-github-release
+spec:
+  params:
+    - description: The release name to target
+      name: release-name
+      type: string
+    - description: The mirrored release pullspec to target
+      name: mirrored-release-pullspec
+      type: string
+    - description: The Kubernetes secret containing a GitHub token
+      name: github-token-secret-name
+      type: string
+      default: gh-token
+    - description: The secret key name for the GitHub token
+      name: github-token-secret-key
+      type: string
+      default: zzlotnik-gh-okd-token
+    - description: The OS release info from the OS container
+      name: os-release
+      type: string
+    - description: The GPG key ID to sign the release
+      name: gpg-key-id
+      type: string
+      default: okd@not.real.go.away
+    - description: The Kube secret name containing the GPG signing key
+      name: gpg-secret-name
+      type: string
+      default: release-signing-gpg-key
+    - description: The Kube secret key containing the GPG signing key
+      name: gpg-secret-key-name
+      type: string
+      default: fake-okd-signing-key.key
+    - description: The GitHub org/repo to release to
+      name: github-org-repo
+      type: string
+      default: cheesesashimi/okd-release-test
+  steps:
+    - image: "image-registry.openshift-image-registry.svc:5000/default/tekton-worker:latest"
+      name: "create-github-release"
+      env:
+        - name: RELEASE_NAME
+          value: $(params.release-name)
+        - name: MIRRORED_RELEASE_PULLSPEC
+          value: $(params.mirrored-release-pullspec)
+        - name: OS_RELEASE
+          value: $(params.os-release)
+        - name: GH_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: $(params.github-token-secret-name)
+              key: $(params.github-token-secret-key)
+      script: |
+        #!/usr/bin/env bash
+        set -euxo pipefail
+
+        # Retrieve the release info, including the pullspecs
+        release_pullspecs="$(oc adm release info --pullspecs "$MIRRORED_RELEASE_PULLSPEC")"
+
+        # Load the release notes template from the ConfigMap
+        release_notes_template="$(cat /var/release-notes-template/release-notes-template.txt)"
+
+        # Substitute the template placeholder for the release info
+        release_notes="${release_notes_template/release_pullspec/$release_pullspecs}"
+
+        # Read the PRETTY_NAME from the /etc/os-release file contents provided
+        pretty_name="$(echo "$OS_RELEASE" | grep 'PRETTY_NAME')"
+        pretty_name="${pretty_name/PRETTY_NAME=\"/}"
+        pretty_name="${pretty_name/\"/}"
+        # Replace the machine-os line with the PRETTY_NAME content
+        release_notes="$(echo "$release_notes" | sed "s/machine-os .*$/machine-os $pretty_name/g")"
```

Comment:
Not sure why would we want to mutate release notes?

Reply:
This has to do with [these instructions](https://github.com/okd-project/okd-scos):

> For now, manually replace the line starting with machine-os under Component Versions: specifying the OS version in the output above which currently still points to the FCOS machine-os-content that is also present in the payload (although unused), with the $PRETTY_NAME of SCOS' /etc/os-release

There's probably a better way to do this though.

Reply:
Okay, we can follow up on that later

Reply:
This has been fixed, the scos image is now additionally tagged as `machine-os-content`